src/linear_algebra/cross_product.lean

```diff
@@ -0,0 +1,102 @@
+/-
+Copyright (c) 2021 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak, Kyle Miller, Eric Wieser
+-/
+
+import linear_algebra.bilinear_map
+import linear_algebra.matrix.determinant
+import linear_algebra.vectors
+
+/-!
+# Cross products
+
+This module defines the cross product of vectors in $R^3$ for $R$ a commutative ring.
+
+## Main definitions
+
+* `cross_product` is the cross product of pairs of vectors in $R^3$.
+
+## Main results
+
+* `triple_product_eq_det`
+
+## Notation
+
+The locale `vectors` gives the following notation:
+
+* `×₃` for the cross product
+
+## Tags
+
+crossproduct
+-/
+
+open_locale vectors
+
+variables {R : Type*} [comm_ring R]
+
+/-- The cross product of two vectors in $R^3$ for $R$ a commutative ring. -/
+def cross_product : (fin 3 → R) →ₗ[R] (fin 3 → R) →ₗ[R] (fin 3 → R) :=
+begin
+  apply linear_map.mk₂ R (λ (a b : fin 3 → R),
+    ![ (a 1)*(b 2) - (a 2)*(b 1) ,
+       (a 2)*(b 0) - (a 0)*(b 2) ,
+       (a 0)*(b 1) - (a 1)*(b 0) ]);
+  intros;
+  simp only [vec3_add,
+    pi.add_apply, algebra.id.smul_eq_mul, matrix.smul_cons, matrix.smul_empty, pi.smul_apply];
+  apply vec3_eq;
+  ring,
+end
```

Comment:
style:
```suggestion
begin
  apply linear_map.mk₂ R (λ a b : fin 3 → R,
    ![a 1 * b 2 - a 2 * b 1,
      a 2 * b 0 - a 0 * b 2,
      a 0 * b 1 - a 1 * b 0]);
  intros;
  simp only [vec3_add, pi.add_apply, smul_eq_mul, matrix.smul_cons, matrix.smul_empty,
    pi.smul_apply];
  apply vec3_eq;
  ring,
end
```

Reply:
I don't like this particular suggestion. As for the first part, I find it more readable with the redundant parentheses. As for the second part, since the line was too long and so it had to be split, I decided to put the line break after `vec3_add` because this is the only thing that was seeded for `simp` ; the remaining lemmata were autogenerated in order to use `simp only`.

Reply:
> As for the first part, I find it more readable with the redundant parentheses

I think this is due to you having prior experience with languages with different precedence rules. These parentheses are not idiomatic in lean.

> As for the second part, since the line was too long and so it had to be split ...

That's fine, my main point there was that I also replaced one of the lemmas.